includes/controllers/class-message.php

```diff
@@ -67,6 +74,50 @@ public function __construct( $args = [] ) {
 		parent::__construct( $args );
 	}
 
+    /**
+     * Blocks user.
+     *
+     * @param WP_REST_Request $request API request.
+     * @return WP_Rest_Response
+     */
+    public function block_user( $request ) {
+
+        // Check authentication.
+        if ( ! is_user_logged_in() ) {
+            return hp\rest_error( 401 );
+        }
+
+        // Get blocked users.
+        $blocked_users = (array) get_user_meta( get_current_user_id(), hp\prefix( 'blocked_users' ), true );
+
+        // Get blocking user ID.
+        $user_id = $request->get_param( 'user_id' );
+
+        if ( in_array( $user_id, $blocked_users, true ) ) {
+
+            // Remove user from blocked list.
+            unset( $blocked_users[ array_search( $user_id, $blocked_users ) ] );
+        } else {
+
+            // Add user to blocked list.
+            $blocked_users[] = $user_id;
+        }
+
+        $blocked_users = array_filter( $blocked_users );
+
+        if ( empty( $blocked_users ) ) {
```

Comment:
It's ok to use ! in cases like this, less brackets and nesting

Reply:
As you can see below, it uses both if-else conditions in this case..